coop_account_check_deposit/models/account_deposit.py

```diff
@@ -0,0 +1,150 @@
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class AccountCheckDeposit(models.Model):
+    _inherit = "account.check.deposit"
+
+    name = fields.Char(string='Name', required=True,
+                       states={'done': [('readonly', '=', True)]},)
+    destination_journal_id = fields.Many2one(
+        'account.journal',
+        string="Destination Journal",
+        required=True, states={'done': [('readonly', '=', True)]},)
+
+    @api.model
+    def _prepare_account_move_vals(self, deposit):
+        date = deposit.deposit_date
+        if deposit.destination_journal_id.sequence_id:
+            move_name =\
+                deposit.destination_journal_id.sequence_id.with_context(
+                    ir_sequence_date=date).next_by_id()
+        else:
+            raise UserError(_(
+                'Please define a sequence on the destination journal.'))
+        move_vals = {
+            'journal_id': deposit.destination_journal_id.id,
+            'date': date,
+            'name': move_name,
+            'ref': deposit.name,
+        }
+        return move_vals
+
+    @api.model
+    def _prepare_move_line_vals(self, line, deposit):
+        # replace assert is raise
+        if line.debit <= 0:
+            raise UserError(_(
+                'Debit must have a value'))
+        res = super(AccountCheckDeposit, self). \
+            _prepare_move_line_vals(
+            line)
+        res['name'] = _('%s - Ref. Check %s') % \
+            (deposit.name, line.ref or line.name or '')
+        return res
+
+    @api.model
+    def _prepare_counterpart_move_lines_vals(
+            self, deposit, total_debit, total_amount_currency):
+        res = super(AccountCheckDeposit, self). \
+            _prepare_counterpart_move_lines_vals(
+            deposit, total_debit, total_amount_currency)
```

Comment:
You can probably do this and keep travis happy:
```suggestion
        res = super()._prepare_counterpart_move_lines_vals(
            deposit, total_debit, total_amount_currency)
```